service/src/main/java/bio/terra/cbas/util/BearerTokenFilter.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.cbas.util;
+
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+/**
+ * Servlet filter that inspects the incoming request for an Authorization: Bearer ... token, and
+ * saves any token it finds into the current thread-local RequestContextHolder.
+ *
+ * <p>Note that this filter does not validate or inspect the token; it just extracts it from the
+ * request, allowing it to be sent as-is from CBAS to other services such as Sam.
+ *
+ * <p>Adapted from <a
+ * href="https://github.com/DataBiosphere/terra-workspace-data-service/blob/b479672d36545d0b45766b3e0e09564d077a19b8/service/src/main/java/org/databiosphere/workspacedataservice/sam/BearerTokenFilter.java">WDS</a>
+ */
+@Component
```

Comment:
Since I found this being a `Component` a little confusing too... could you add a quick comment along the lines of how Spring uses these `Filter` components to populate the request attributes with this sort of pseudo-attribute?

And TOL for the same of simplicity... you could probably also have had this be a static utility function that takes the list of attributes as an argument and returns the auth token, without any need for all this magic filter component instantiation, like
```
public static String getAuthToken(RequestAttributes attributes) { ... }
```

Reply:
My thought here was that the filter may be useful for logging purposes later down the line, since it's automatically applied once to every incoming request (whereas the static utility function would be called each time we get the user token, and I'm not sure about the consistency across requests). I think I may be looking ahead too much though, since I imagine logging may end up being more complicated and request-dependent.

Reply:
Tried moving into a helper/utility method - the problem that arises is suddenly we are dealing with `request` objects again, since headers are not in the attributes. I think this `Filter` may be a necessary evil if we want to restrict the number of somewhat opaque request objects being passed around. I will add/update comments for better context